File: 11.container-with-most-water.py
# @leet start
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)

        l = 0
        r = n - 1
        most = 0

        while l < r:
            water = (r - l) * min(height[l], height[r])
            most = max(water, most)

            if height[l] < height[r]:
                l += 1
            else:
                r -= 1

        return most


# @leet end

File: 11.container-with-most-water.java

// @leet imports start
import java.util.*;
import java.math.*;
// @leet imports end

// @leet start
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int l = 0;
        int r = n - 1;
        int ans = 0;
        while (l < r) {
            int hl = height[l];
            int hr = height[r];
            int water = Math.min(hl, hr) * (r - l);
            ans = Math.max(ans, water);
            if (hl < hr) {
                l++;
            } else {
                r--;
            }
        }
        return ans;

    }
}
// @leet end
